### ISSUE TITLE
Enhance Train Ticket Booking API with RAC and Waiting List Features

### ISSUE THREAD

**Issue by RameshKanish** [Mayank2808sharma/IRCTC-backend#23]
The current version of our open-source project's Train Ticket Booking API lacks support for features like RAC (Reservation Against Cancellation) and Waiting List tickets. Additionally, there is no restriction on the number of tickets a user can book, potentially leading to abuse by bulk bookers.

Proposed Solution:

RAC and Waiting List Implementation: Introduce functionality to support RAC and Waiting List tickets in the Train Ticket Booking API. This will enhance the user experience by providing flexibility in ticket booking, especially during peak times when seats are limited.

Ticket Booking Limit: Implement a restriction on the number of tickets a user can book in a single transaction. Currently, users can book an unlimited number of tickets, which may lead to unfair booking practices and inconvenience for other users. Setting a maximum limit of six tickets per user per transaction will ensure fair distribution of available seats.

I would like to proceed with working on this issue 

Would You assign this issue to me 
contribution for gssoc 2024

**Comment by Mayank2808sharma** [comment]
@RameshKanish
Go ahead I have assigned the issue to you

**Comment by sneharini** [comment]
Hey @Mayank2808sharma,
Would it be possible for me to collaborate on this matter with @RameshKanish? I'm relatively new to contributing to open source, so having someone with experience to work alongside would be incredibly beneficial.

**Comment by RameshKanish** [comment]
> Hey @Mayank2808sharma, Would it be possible for me to collaborate on this matter with @RameshKanish? I'm relatively new to contributing to open source, so having someone with experience to work alongside would be incredibly beneficial.



yes @Mayank2808sharma  could you assign the same task to her (@sneharini ) we both work together to contribute to the API.
